File: src/main/java/com/wisenut/infrastructures/repository/kakao/KakaoMapOfferRepository.java
package com.wisenut.infrastructures.repository.kakao;

import com.wisenut.domain.model.kakaomap.KaKaoMapOffer;
import org.springframework.data.jpa.repository.JpaRepository;

public interface KakaoMapOfferRepository extends JpaRepository<KaKaoMapOffer, Long>, KakaoMapOfferRepositoryCustom{
    KaKaoMapOffer findByType(String type);
}

File: src/main/java/com/wisenut/domain/model/kakaomap/KakaoMapOfferRepository.kt
package com.wisenut.domain.model.kakaomap

import org.springframework.data.jpa.repository.JpaRepository
import com.wisenut.infrastructures.repository.kakao.KakaoMapOfferRepositoryCustom

interface KakaoMapOfferRepository : JpaRepository<KaKaoMapOffer?, Long?>, KakaoMapOfferRepositoryCustom {
    fun findByType(type: String?): KaKaoMapOffer?
}